narwhals/_spark_like/dataframe.py

```diff
@@ -312,3 +314,57 @@ def join(
         return self._from_native_frame(
             self_native.join(other, on=left_on, how=how).select(col_order)
         )
+
+    def explode(self: Self, columns: str | Sequence[str], *more_columns: str) -> Self:
+        from narwhals.exceptions import InvalidOperationError
+
+        dtypes = import_dtypes_module(self._version)
+
+        to_explode = (
+            [columns, *more_columns]
+            if isinstance(columns, str)
+            else [*columns, *more_columns]
+        )
+        schema = self.collect_schema()
+        for col_to_explode in to_explode:
+            dtype = schema[col_to_explode]
+
+            if dtype != dtypes.List:
+                msg = (
+                    f"`explode` operation not supported for dtype `{dtype}`, "
+                    "expected List type"
+                )
+                raise InvalidOperationError(msg)
+
+        native_frame = self._native_frame
+        column_names = self.columns
+
+        def null_condition(col_name: str) -> Column:
+            return self._F.isnull(col_name) | (self._F.array_size(col_name) == 0)
+
+        if len(to_explode) == 1:
+            return self._from_native_frame(
+                native_frame.select(
+                    *[
+                        self._F.col(col_name).alias(col_name)
+                        if col_name != to_explode[0]
+                        else self._F.explode(col_name).alias(col_name)
+                        for col_name in column_names
+                    ]
+                ).union(
+                    native_frame.filter(null_condition(to_explode[0])).select(
+                        *[
+                            self._F.col(col_name).alias(col_name)
+                            if col_name != to_explode[0]
+                            else self._F.lit(None).alias(col_name)
+                            for col_name in column_names
+                        ]
+                    )
+                )
+            )
+
+        msg = (
+            "Exploding on multiple columns is not supported with SparkLike backend since "
+            "we cannot guarantee that the exploded columns have matching element counts."
+        )
+        raise NotImplementedError(msg)
```

Comment:
In a way... this is so sad since I had a great way of doing it! But cannot verify that the element counts are same without triggering a collect